src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -151,6 +155,41 @@ public PersistentDataStore() {
     return messages;
   }
 
+  /**
+   * Loads all User objects from the Datastore service and returns them in a List.
+   *
+   * @throws PersistentDataStoreException if an error was detected during the load from the
+   *     Datastore service
+   */
+  public List<Hashtag> loadHashtags() throws PersistentDataStoreException {
+
+    List<Hashtag> hashtags = new ArrayList<>();
+
+    // Retrieve all hashtags from the datastore.
+    Query query = new Query("chat-hashtags");
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      try {
+        String tagName = (String) entity.getProperty("tag_name");
+        List<String> messageIds = new ArrayList( (Collection<String>) entity.getProperty("uuid_list"));
+        List<UUID> messageIdsList = new ArrayList<UUID>();
```

Comment:
```java
List<UUID> messageIDs = messagesIds.stream().map(UUID::fromString).collect(Collectors.toList());
```

As an aside, we avoid writing "list" or "map" at the end of variable names unless absolutely necessary. For lists, sets, etc. we simply make it plural (i.e. messageIds) and for maps we do x by y (messageIdsByName, for example).